### ISSUE TITLE
No Signin id passed - SAML-P / Kentor

### ISSUE THREAD

**Issue by JulienC75** [IdentityServer/IdentityServer3#1532]
Hi,

I'm trying to use Kentor with Identity server but I get this message in logs: "No Signin id passed"
And IdSvr displays "There is an error determining which application you are signing into. Return to the application and try again."

I've seen similar issues using WS-Federation but I can't figure out how to solve the issue in my case.

Moreover, as soon as I activate the Kentor authentication, the general behavior of IDSRV become strange: 
- login with the default authentication works, but the authorizations are not correctly handled (I can't access to some controllers I have)
- logout do not work correctly (no message saying you correcly logout, no redirection link)

When I deactivate Kentor authentication, the behavior come back to normal.

This is the method to configure identity providers:

public static void ConfigureIdentityProviders(IAppBuilder app, string signInAsType)
{
	var authServicesOptions = new KentorAuthServicesAuthenticationOptions(false)
	{
		SPOptions = new SPOptions
		{
			EntityId = new EntityId("https://localhost/IdentityServer/AuthServices/"),
		},
		SignInAsAuthenticationType = signInAsType,
		AuthenticationType = "saml2p",
		Caption = "SAML2p",
	};

	authServicesOptions.IdentityProviders.Add(
		new IdentityProvider(
			new EntityId("https://entityid.idp.test"), authServicesOptions.SPOptions)
		{
			MetadataUrl=new Uri("https://localhost/SimpleSAMLPHPIDP/saml2/idp/metadata.php"),
			LoadMetadata = true,
		});
	app.UseKentorAuthServicesAuthentication(authServicesOptions);
}

and the startup

 var idsrvOption = new IdentityServerOptions
            {
                SiteName = "Identity Server",
                Factory = FactoryHelper.CreateIdSrvServiceFactory(),
				SigningCertificate = LoadCertificate(),

                CorsPolicy = CorsPolicy.AllowAll,

				AuthenticationOptions = new Thinktecture.IdentityServer.Core.Configuration.AuthenticationOptions()
				{
					IdentityProviders = ConfigureIdentityProviders, 
				},
				
                LoggingOptions = new LoggingOptions
                {
                    EnableWebApiDiagnostics = true,
                    WebApiDiagnosticsIsVerbose = true,
                    EnableHttpLogging = true,
                },

                EventsOptions = new EventsOptions
                {
                    RaiseSuccessEvents = true,
                    RaiseErrorEvents = true,
                    RaiseFailureEvents = true,
                    RaiseInformationEvents = true
				},
            };

app.Map("/identity", idsrvApp => 
            { 
                idsrvApp.UseIdentityServer(idsrvOption);
                idsrvApp.UseWsFederationPlugin(wsFedOption);
            });

 app.UseCookieAuthentication(new CookieAuthenticationOptions
            {
                AuthenticationType = "Cookies"
            });
            
            app.UseOpenIdConnectAuthentication(new OpenIdConnectAuthenticationOptions
            {
                Authority = "https://localhost/IdentityServer/identity",
				ClientId = "db86d6ea-567f-4df3-b340-c21f88652182",
                RedirectUri = "https://localhost/IdentityServer/",
                Scope = "openid profile roles identity",
                ResponseType = "id_token token",

                SignInAsAuthenticationType = "Cookies",
                UseTokenLifetime = false,

                Notifications = new OpenIdConnectAuthenticationNotifications
                {
                    SecurityTokenValidated = async n =>
                    {
                        var id = n.AuthenticationTicket.Identity;
                        // get userinfo data
                        var nid = new ClaimsIdentity(id.AuthenticationType, id.Name, Constants.ClaimTypes.Role);
                        // get userinfo data
                        var userInfoClient = new UserInfoClient(new Uri(n.Options.Authority + "/connect/userinfo"), n.ProtocolMessage.AccessToken);
                        var userInfo = await userInfoClient.GetAsync();
                        userInfo.Claims.ToList().ForEach(ui => nid.AddClaim(new Claim(ui.Item1, ui.Item2)));

                        // keep the id_token for logout
                        nid.AddClaim(new Claim("id_token", n.ProtocolMessage.IdToken));
                        // add access token for sample API
                        nid.AddClaim(new Claim("access_token", n.ProtocolMessage.AccessToken));
                        // keep track of access token expiration
                        nid.AddClaim(new Claim("expires_at", DateTimeOffset.Now.AddSeconds(int.Parse(n.ProtocolMessage.ExpiresIn)).ToString()));

                        n.AuthenticationTicket = new AuthenticationTicket(nid, n.AuthenticationTicket.Properties);
                    },
                    RedirectToIdentityProvider = async n =>
                    {
                        if (n.ProtocolMessage.RequestType == OpenIdConnectRequestType.LogoutRequest)
                        {
                            var idTokenHint = n.OwinContext.Authentication.User.FindFirst("id_token");
							if(idTokenHint != null)
								n.ProtocolMessage.IdTokenHint = idTokenHint.Value;
                        }
                    }
                }
            });

Any idea what am I doing wrong?

Julien

**Comment by leastprivilege** [comment]
Seems you are mixing idsrv and the client in the same web app - i would separate them.

**Comment by JulienC75** [comment]
After reading other posts I also tried this, however it doesn't solve the issue.

I removed app.UseOpenIdConnectAuthentication from idsrv and created a new client application with the following code (just changed clientid to a new one). 

Without kentor it works. When I add kentor, it raises the same error.

app.UseOpenIdConnectAuthentication(new OpenIdConnectAuthenticationOptions
			{
				Authority = "https://localhost/IdentityServer/identity",
				**ClientId = "ClientApplication",**
				**RedirectUri = "http://localhost/ClientApplication/",**
				*... then same code as previous post...*
			});
